librocksdb_sys/crocksdb/c.cc

```diff
@@ -5109,4 +5115,87 @@ void ctitandb_options_set_blob_run_mode(ctitandb_options_t* options,
   options->rep.blob_run_mode = static_cast<TitanBlobRunMode>(mode);
 }
 
+/* TitanReadOptions */
+struct ctitandb_readoptions_t {
+  TitanReadOptions rep;
+};
+
+ctitandb_readoptions_t* ctitandb_readoptions_create() { 
+  return new ctitandb_readoptions_t; 
+}
+
+void ctitandb_readoptions_destroy(ctitandb_readoptions_t* opts) { 
+  delete opts; 
+}
+
+uint64_t ctitandb_readoptions_key_only(ctitandb_readoptions_t* opts) {
+  return opts->rep.key_only;
+}
+
+void ctitandb_readoptions_set_key_only(ctitandb_readoptions_t* opts,
+                                        bool v) {
+  opts->rep.key_only = v;
+}
+
+crocksdb_iterator_t* ctitandb_create_iterator(
+    crocksdb_t* db,
+    const crocksdb_readoptions_t* options,
+    const ctitandb_readoptions_t* titan_options) {
+  crocksdb_iterator_t* result = new crocksdb_iterator_t;
+  if (titan_options == nullptr) {
+    result->rep = db->rep->NewIterator(options->rep);
+  } else {
+    *(ReadOptions*)&titan_options->rep = options->rep;
```

Comment:
Why override them with `options`?

Reply:
the definition is like that `public TitanReadOptions: public ReadOptions`
This is to perform the copy assignment of the `ReadOptions` part of `TitanReadOptions`.

Reply:
if follows the how `DBOptions` does, see https://github.com/pingcap/rust-rocksdb/pull/309/files/02a3b0ad761c0d18326ef0f274f7c3fe49a7a5ab#diff-733c27ed8699d3baa6cbff3ac91a464bR4976